lib/pages/add_bridge/add_bridge.dart

```diff
@@ -628,6 +799,90 @@ class BotController extends State<AddBridge> {
   // 📌 ************************** Messenger & Instagram **************************
   // 📌 ***********************************************************************
 
+  Future<void> bridgeLoginApi(
+      BuildContext context,
+      WebviewCookieManager cookieManager,
+      ConnectionStateModel connectionState,
+      SocialNetwork network,
+      ) async {
+    final flowID = network.flowId;
+    final bridgePathIdentifier  = getBridgePath(network);
+    final userId = client.userID;
+
+    // Step 1: Start the login process
+    final loginStartUrl = 'https://matrix.$serverUrl/_matrix/$bridgePathIdentifier/provision/v3/login/start/$flowID?user_id=$userId';
+
+    try {
+      // Initiate the login process
+      final startResponse = await requestPost(loginStartUrl);
+
+      if (startResponse.statusCode == 200) {
+        final startData = jsonDecode(startResponse.body);
+        final loginId = startData['login_id'];
+        final stepType = startData['type'];
+        final stepId = startData['step_id'];
+
+        // Update state if there's a display message
+        if (stepType == 'display_and_wait') {
+          connectionState.updateConnectionTitle(
+            startData['instructions'] ?? 'Waiting for user action...',
+          );
+        }
+
+        // Step 2: If the next step is user input or cookies, prepare to submit cookies
+        if (stepType == 'user_input' || stepType == 'cookies') {
+          // Retrieve cookies
+          final gotCookies = await cookieManager.getCookies(network.urlRedirect);
+          final formattedCookieString = formatCookiesToJsonApi(gotCookies);
+
+          // Submit cookies to the login process step
+          final stepUrl = 'https://matrix.$serverUrl/_matrix/$bridgePathIdentifier/provision/v3/login/step/$loginId/$stepId/cookies?user_id=$userId';
+
+          final stepResponse = await requestPost(stepUrl, body: formattedCookieString);
+
+          if (stepResponse.statusCode == 200) {
+            final stepData = jsonDecode(stepResponse.body);
+            if (stepData['type'] == 'complete') {
+              setState(() => network.updateConnectionResult(true));
+              if (kDebugMode) {
+                print('Login successful for ${network.name}');
+              }
+            } else {
+              network.setError(true);
+              if (kDebugMode) {
+                print('Unexpected response: ${stepData['type']}');
+              }
+            }
+          } else {
+            network.setError(true);
```

Comment:
Error messages are not displayed to user. Try throwing exceptions and then catching them in one place. Give the user context to understand what went wrong